openedx/core/djangoapps/programs/tasks/v1/tests/test_tasks.py

```diff
@@ -165,12 +165,20 @@ def test_award_program_certificate(self):
         """
         test_username = 'test-username'
         test_client = EdxRestApiClient('http://test-server', jwt='test-token')
+
         httpretty.register_uri(
             httpretty.POST,
             'http://test-server/user_credentials/',
         )
+
         tasks.award_program_certificate(test_client, test_username, 123)
-        self.assertEqual(httpretty.last_request().body, json.dumps({'program_id': 123, 'username': test_username}))
+
+        expected_body = {
+            'username': test_username,
+            'credential': {'program_id': 123},
+            'attributes': []
+        }
+        self.assertEqual(httpretty.last_request().body, json.dumps(expected_body))
```

Comment:
Cool. I imagine that `httpretty` is using the same `json.dumps` internally, and dictionaries have a deterministic iteration order, so it'll all Just Work when two bits of Python code are communicating through JSON.